readme.md

```diff
@@ -1,57 +1,79 @@
 # Mannish the Manly Mediator
 
-Inspired by Nicholas Zakas's talk on "[Creating A Scalable JavaScript Application Architecture](http://youtu.be/b5pFv9NB9fs)", this is a mediator to let modules communicate without tight coupling - for whatever "module" means to you.
+![](mannish.jpg)
 
-# Usage
+This is a fairly specific implementation of the mediator pattern - given a string name, it calls a function and returns a promise with the functions response.
 
-## Create a new app context
+It accomplishes the same goals as many dependency injection libraries, but focuses only on functions, and always returning asynchronous responses.
 
-```js
-var mannish = require('mannish')
-
-var appContext = mannish()
-```
+Originally inspired by Nicholas Zakas's talk on "[Creating A Scalable JavaScript Application Architecture](http://youtu.be/b5pFv9NB9fs)", this library fills the role of what he called the "sandbox" more closely than it serves the purpose he had for the mediator pattern.
 
-## Subscribe, adding yourself as a provider
+If you want a similar library to be used for broadcasting (something Mannish supported in older versions), I would recommend something like:
 
 ```js
-appContext.subscribe('unixTimestamp', function(someDate, cb) {
-	cb(null, Math.round(someDate.getTime() / 1000))
-})
+function makeBroadcaster() {
+	const emitter = new EventEmitter()
+
+	return {
+		emit(...args) {
+			emitter.emit(...args)
+		},
+		on(name, listener) {
+			emitter.on(name, listener)
+
+			return function removeListener() {
+				emitter.removeListener(name, listener)
+			}
+		}
+	}
+}
```

Comment:
Anyone trying to skim the readme would probably take this code block to be example usage, and then be confused.